### ISSUE TITLE
Improve numerical error message clarity

### ISSUE THREAD

**Issue by yebai** [TuringLang/AdvancedHMC.jl#110]
The current numerical error message looks like
```julia
┌ Warning: The current proposal will be rejected due to numerical error(s).
│   isfiniteθ = true
│   isfiniter = false
│   isfiniteℓπ = false
│   isfiniteℓκ = false
```

Maybe the following is slightly more neat and clearer?

```julia

┌ Warning: The current proposal will be rejected due to numerical error(s).
│   isfinite.((θ,r,ℓπ,ℓκ)) = (true, false, false, false)
```

**Comment by xukai92** [comment]
Related issue https://github.com/TuringLang/Turing.jl/issues/974#issuecomment-555395341

**Comment by yebai** [comment]
Closed by #136
